### ISSUE TITLE
Moving messages without folder hierarchy on host 2

### ISSUE THREAD

**Issue by elvisef** [imapsync/imapsync#229]
Hello Gilles.

I've send You an email, but I suppose You didn't receive it yet. So Ill write here, cause it's important to me.

Generally, I have a question about imapsync.
 
I've already looked at these information:
https://imapsync.lamiral.info/README
https://imapsync.lamiral.info/FAQ.d/FAQ.Folders_Mapping.txt
https://imapsync.lamiral.info/FAQ.d/FAQ.Folders_Selection.txt
 
but I did not find solution.
 
 
Example folder structure:
 
**host1:**

INBOX
```
 -- test 1
 -- test 2
     -- test888
 -- test908
```
 
**host2:**

INBOX
 
 
Now I want to transfer all emails with a header "FROM @ebay.com" from host1 to host2 but... without creating the folder structure on host2.
 
I thought will be simple, but the problem is if I have --folderrec INBOX (I am using --dry option to test for now), folders are being created on host2 with the whole hierarchy:
 
Creating folder [INBOX/[XXX]/Finanse/Fundusze/testing] on host2
Creating folder [INBOX/[XXX]/Finanse/Fundusze] on host2
Creating folder [INBOX/[XXX]/Finanse] on host2
 
etc...
 
How to: 1) search for certain messages (using header for instance FROM), then 2) using the whole INBOX (recursively) copy those messages to host 2  3) to folder INBOX 4) without creating the whole structure from host1 ?
 
Because as I see if the messages on host 1 are in certain subdirectories, all these subdirectories are being created on host 2...
 
Tried mixing options but no luck ; /

**EDIT** 23.08.2020

Alright, I think this makes it clear:

--regextrans2 "s/.*/INBOX2/"

But wondering, why cannot use for example this:

--folderrec INBOX --f1f2 INBOX=INBOX2 - why this recreates the whole hierarchy and the previous solution not ?

**Comment by gilleslamiral** [comment]
> --folderrec INBOX --f1f2 INBOX=INBOX2 - why this recreates the whole hierarchy and the previous solution not ?

Because --f1f2 INBOX=INBOX2  applies strictly to one folder only, the one on the left of the = character.

Final:

    imapsync ... --search "FROM foo@example.com" --folderrec INBOX  --regextrans2 "s/.*/INBOX/"

**Comment by elvisef** [comment]
Alright. So I have a question about **"--regextrans2 "s/.*/INBOX/"**
 
**.*** - does it mean here "INBOX" from "folderrec" argument ?
 
What if I wanted to transfer **"test 2"** folder only, recursively of course ? This would transfer **"test 2"** and **"test888"** (as in my sample structure). How would You prepare the terminal options ?
 
Something like this: **--folderrec "INBOX/[test 2]"** ? Tried this but did not work.
 
**The point is I want to transfer some emails from one of folders in the whole structure (and this one folder has some folders in it, and tthese folders has some in them etc..).**

**Comment by gilleslamiral** [comment]
> I have a question about "--regextrans2 "s/.*/INBOX/"
> does it mean here "INBOX" from "folderrec" argument ?

No. INBOX in "--regextrans2 "s/.*/INBOX/" is the destination folder on host2.

Please include some log from the imapsync output as I don't understand if the folder you deal with is named "test 2" or "INBOX/[test 2]". I don't know either the separators, nor the prefixes, nor the OS context, these things are relevant me to give the right answer strait.

To transfer "test 2" folder recursively, use:

    imapsync ... --folderrec "test 2"

https://imapsync.lamiral.info/FAQ.d/FAQ.Folders_Selection.txt
...
````
Q. How to make option --folder "MyFolder" be recursive?

Two solutions:

R1. Use

--folderrec "MyFolder"

R2. Use --include "^MyFolder"
   Then the folder "MyFolder" and all its subfolders will be handled
   and only them.
````

**Comment by elvisef** [comment]
> > I have a question about "--regextrans2 "s/.*/INBOX/"
> > does it mean here "INBOX" from "folderrec" argument ?
> 
> No. INBOX in "--regextrans2 "s/.*/INBOX/" is the destination folder on host2.
> 
Here I've asked about ".*", not "INBOX" : )

Alright, I will give You LOG from the real example, I want to achieve.

I am using this command (sample only):
`imapsync --host1  --user1  --password1  --host2  --user2  --password2  --ssl1 --ssl2 --port1 993 --port2 993 --search2 "TO destination.email@jerry.pl" --nosyncacls --syncinternaldates --skipemptyfolders --nomixfolders --nousecache --noreleasecheck --useuid --nofoldersizes --nocheckselectable --nofoldersizesatend --no-modulesversion --folderrec Praca --regextrans2 "s/.*/Sent/" --dry`

Log attached.
[2021_03_16_14_24_20_524.txt](https://github.com/imapsync/imapsync/files/6149342/2021_03_16_14_24_20_524.txt)

What do I want to do ?

I want to transfer --> move messages on THE SAME server, from one folder to another.
A folder named "Praca" has some emails, that should be in the folder "Sent", because I failed copying them from another server and it just happened that there they are.

So I want to move them from Praca, to Sent. - On the same server. Just change folders for them.
If I am using "TO nameofemailaccount@domain" it does not find anything.

Am I using wrong folder name convention ?

**Comment by gilleslamiral** [comment]
a) The log file you sent ends fast with:

     Host1 failure: Error login on [imap.poczta.onet.pl] with user [jmar@buziaczek.pl] auth [LOGIN]: 2 NO [AUTHENTICATIONFAILED] Authentication failed.

It's useless in regard of your previous demand, it doesn't login on host1.

b) 
> I want to transfer --> move messages on THE SAME server, from one folder to another.

I guess you mean the same account instead of same server. Moving messages is done with the --delete1 option. There is no move command in imap. Move is implemented within imapsync as a copy+delete, so add the --delete1 option.

To move all messages from the folder Praca to the folder Sent, use:

    imapsync .... --folder Praca --f1f2 Praca=Sent --delete1 

To move all messages from the folder Praca and all messages from all the subfolders of the folder Praca folder, to the same folder Sent, use:

  imapsync .... --folderrec  Praca --regextrans2 s/.*/Sent/ --delete1
